micrometer-core/src/main/java/io/micrometer/core/instrument/internal/DefaultLongTaskTimer.java

```diff
@@ -156,25 +156,23 @@ public HistogramSnapshot takeSnapshot() {
             List<CountAtBucket> countAtBuckets = new ArrayList<>(buckets.size());
 
             SampleImpl priorActiveTask = null;
-            int i = 0;
+            int count = 0;
 
             Iterator<SampleImpl> youngestToOldest = activeTasks.descendingIterator();
             while (youngestToOldest.hasNext()) {
                 SampleImpl activeTask = youngestToOldest.next();
-                i++;
-                if (bucket != null) {
-                    if (activeTask.duration(TimeUnit.NANOSECONDS) > bucket) {
-                        countAtBuckets.add(new CountAtBucket(bucket, i - 1));
-                        bucket = buckets.pollFirst();
-                    }
+                while (bucket != null && activeTask.duration(TimeUnit.NANOSECONDS) > bucket) {
```

Comment:
Is `activeTask.duration(TimeUnit.NANOSECONDS)` recalculated in every iteration of this loop (as time passes while we iterate so as the duration)? Is it ok if it does?

Reply:
It's calculated based on the configured `Clock`, which for this test is the `MockClock`, so it isn't changing unless explicitly updated.

Reply:
Yes, that could result in some inconsistency. Let me update.

Reply:
On thinking about it more, I don't think it would result in an inconsistent histogram. Though to keep in the spirit of a "snapshot" we should probably get the duration of all tasks early and use those rather than always using the latest duration.

Reply:
Updated. Please check again.